src/font.rs

```diff
@@ -155,8 +155,8 @@ impl<'a> PdfFont<'a> {
 
     /// Creates a new [PdfFont] from the given given built-in font argument.
     #[inline]
-    pub fn new_built_in(document: &'a PdfDocument<'a>, font: PdfFontBuiltin) -> Self {
-        let mut result = Self::from_pdfium(
+    pub fn new_built_in(document: &PdfDocument<'a>, font: PdfFontBuiltin) -> PdfFont<'a> {
```

Comment:
all changes here are just relaxing lifetimes a bit

I'm not claiming this is the minimal change - there might be a slightly smaller diff which also makes this work

with no change, you see that creating a font holds onto the document ref



```
pdfium-render % cargo run --example attachments
    Blocking waiting for file lock on build directory
   Compiling pdfium-render v0.8.1 (/Users/bduffield/Git/pdfium-render)
error[E0597]: `document` does not live long enough
  --> examples/attachments.rs:23:33
   |
23 |             &PdfFont::helvetica(&document),
   |                                 ^^^^^^^^^ borrowed value does not live long enough
...
65 | }
   | -
   | |
   | `document` dropped here while still borrowed
   | borrow might be used here, when `document` is dropped and runs the `Drop` code for type `PdfDocument`

error[E0502]: cannot borrow `document` as mutable because it is also borrowed as immutable
  --> examples/attachments.rs:29:5
   |
23 |               &PdfFont::helvetica(&document),
   |                                   --------- immutable borrow occurs here
...
29 | /     document
30 | |         .attachments_mut()
   | |__________________________^ mutable borrow occurs here

...[snip]...
```

Reply:
I believe this might fix https://github.com/ajrcarey/pdfium-render/issues/79?

Reply:
I had the same thought. I have some time set aside tomorrow to run some more tests on this, including making sure it works in WASM.

Reply:
Sadly I don't think this adjustment alone is enough, the fundamental problem of trying to mix mutable and immutable references on PdfDocument remains the same. I believe the only solution is for PdfDocument to take ownership of the PdfFont you want to use and let you retrieve it on demand. I have an idea as to how to do this and I will comment on that further in #79.